formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonStreams.kt

```diff
@@ -47,6 +51,7 @@ internal fun <T> Json.decodeToSequenceByReader(
     deserializer: DeserializationStrategy<T>,
     format: DecodeSequenceMode = DecodeSequenceMode.AUTO_DETECT
 ): Sequence<T> {
+    // Note: no explicit release, as the sequence are lazy and thrown away in an arbitrary manner
```

Comment:
Do I understand correctly that this function permanently "steals" an array from the pool and if `decodeToSequenceByReader` and `decodeByReader` are called in the application, then in fact this is equivalent to creating a new instance of the array for each call to `decodeToSequenceByReader`?

Reply:
Correct

Reply:
Maybe in this case it is better not to use the pool for `decodeToSequenceByReader` so that not affects `decodeByReader`?
At the same time, in any case, each call to `decodeToSequenceByReader` will require the creation of a new array (possibly in another thread), so the total amount of work does not change from this.

Reply:
+1. Looks like 'taking and never returning' in decodeToSequence may negatively affect decodeFromStream

Reply:
Alternatively, you may try to add `lexer.release()` to `JsonIterator.hasNext()` implementation. This may solve the problems at the first sight but may require thorough testing.

Reply:
Although it would still steal buffer if the resulting sequence is used together with things like `.take()`/`limit`, but it's still better than current solution?